containers-contrib-readonly/src/Map/Contrib/Bench.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Map.Contrib.Bench where
+
+import Map
+import Gauge.Main (Benchmark, bench, defaultMain, whnf)
+import Relude hiding (fromList, Map)
+
+benchmark
+  :: forall k. (Enum k, Bounded k, Key k, NFData (Map k Int))
```

Comment:
Instead of requiring `NFData` constraint in this benchmark, you can require maps to have `NFData` instance by adding it in `.hsig` file here:

https://github.com/kowainik/containers-backpack/blob/c84b22f7551bd34183ac76ff20dd4e4fa4cd6f48/containers-sig-readonly/src/Map.hsig#L29-L30

Reply:
Hmm, I see the problem now. The fact that `Map.Prim` doesn't have `NFData` instance really makes things more difficult.